ngraph/frontend/onnx_import/src/op/resize.cpp

```diff
@@ -28,6 +29,124 @@ namespace ngraph
         {
             namespace
             {
+                static const std::unordered_set<std::string> supported_modes = {
+                    "nearest", "linear", "cubic"};
+
+                static const std::unordered_set<std::string> supported_transforms = {
+                    "half_pixel",
+                    "pytorch_half_pixel",
+                    "align_corners",
+                    "asymmetric",
+                    "tf_half_pixel_for_nn"};
+
+                using InterpolateMode = ngraph::op::v4::Interpolate::InterpolateMode;
+                static const std::map<std::string, InterpolateMode> interp_mode_map = {
+                    {"nearest", InterpolateMode::nearest},
+                    {"linear", InterpolateMode::linear_onnx},
+                    {"cubic", InterpolateMode::cubic}};
+
+                using Transform_mode = ngraph::op::v4::Interpolate::CoordinateTransformMode;
+                static const std::map<std::string, Transform_mode> transform_mode_map = {
+                    {"half_pixel", Transform_mode::half_pixel},
+                    {"pytorch_half_pixel", Transform_mode::pytorch_half_pixel},
+                    {"align_corners", Transform_mode::align_corners},
+                    {"asymmetric", Transform_mode::asymmetric},
+                    {"tf_half_pixel_for_nn", Transform_mode::tf_half_pixel_for_nn}};
+
+                using Nearest_mode = ngraph::op::v4::Interpolate::NearestMode;
+                static const std::map<std::string, Nearest_mode> nearest_mode_map = {
+                    {"round_prefer_floor", Nearest_mode::round_prefer_floor},
+                    {"round_prefer_ceil", Nearest_mode::round_prefer_ceil},
+                    {"floor", Nearest_mode::floor},
+                    {"ceil", Nearest_mode::ceil},
+                    {"simple", Nearest_mode::simple}};
+
+                static bool is_supported_str_value(const std::unordered_set<std::string>& modes,
+                                                   const std::string& checked_mode)
+                {
+                    return std::find(modes.begin(), modes.end(), checked_mode) != modes.end();
+                }
+
+                template <typename T>
```

Comment:
I don't think you need a template here. You don't even have to pass the map as an argument since it's a static const in this namespace. Also I think this function's name is too vague - try to change it to something more meaningful explaining what gets converted here.

Reply:
This function was renamed. Next, a template function is needed here and this this template function should have map as an argument, because otherwise we will have three almost identical conversion function with different returned value type and different used maps.

Reply:
Looks like all map contains enums as values, in this case you don't need a template argument here... You can return int64 and after that cast the value to enum.

Reply:
Fixed.